unity-client/Assets/Scripts/MainScripts/DCL/DataStore/Variables/BaseVariable.cs

```diff
@@ -0,0 +1,31 @@
+﻿using System;
+using System.Collections.Generic;
+
+public delegate void Change<T>(T current, T previous);
+
+public class BaseVariable<T> : IBaseVariable<T>, IEquatable<T>
```

Comment:
How come we dont have compilation issues with this? `BaseVariable<T>` type will be duplicated right?